### ISSUE TITLE
(5).toString(); crashes with wasm trap: out of bounds memory access

### ISSUE THREAD

**Issue by phenax** [AssemblyScript/assemblyscript#2735]
### Bug description

```
abort:  in ~lib/rt/itcms.ts(160:16)
Error: failed to run main module `build/debug.wasm`

Caused by:
    0: failed to invoke `my_func`
    1: error while executing at wasm backtrace:
           0: 0x1e19 - <unknown>!~lib/wasi_internal/wasi_abort
           1: 0x1474 - <unknown>!~lib/rt/itcms/visitRoots
           2: 0x16a5 - <unknown>!~lib/rt/itcms/step
           3: 0x17d1 - <unknown>!~lib/rt/itcms/interrupt
           4: 0x1852 - <unknown>!~lib/rt/itcms/__new
           5: 0x20d9 - <unknown>!~lib/util/number/itoa32
           6: 0x1b03 - <unknown>!~lib/number/I32#toString
           7: 0x2054 - <unknown>!assembly/index/my_func
    2: exit with invalid exit status outside of [0..126)

```

### Steps to reproduce

* AssemblyScript file
```ts
// assembly/index.ts
export function my_func(): void {
  (5).toString();
}
```

* Use `@assemblyscript/wasi-shim: v0.1.0`

* Build
```
asc assembly/index.ts --target debug
```

* Run
```
wasmtime build/debug.wasm --invoke my_func
```

### AssemblyScript version

0.27.9

**Comment by MaxGraey** [comment]
Can't [reproduce on playground](https://www.assemblyscript.org/editor.html#IyFvcHRpbWl6ZT1zcGVlZCZydW50aW1lPWluY3JlbWVudGFsCmV4cG9ydCBmdW5jdGlvbiBteV9mdW5jKCk6IHN0cmluZyB7CiAgcmV0dXJuICg1KS50b1N0cmluZygpOwp9CiMhaHRtbAo8dGV4dGFyZWEgaWQ9Im91dHB1dCIgc3R5bGU9ImhlaWdodDogMTAwJTsgd2lkdGg6IDEwMCUiIHJlYWRvbmx5PjwvdGV4dGFyZWE+CjxzY3JpcHQgdHlwZT0ibW9kdWxlIj4KY29uc3QgZXhwb3J0cyA9IGF3YWl0IGluc3RhbnRpYXRlKGF3YWl0IGNvbXBpbGUoKSwgeyAvKiBpbXBvcnRzICovIH0pCmNvbnN0IG91dHB1dCA9IGRvY3VtZW50LmdldEVsZW1lbnRCeUlkKCdvdXRwdXQnKQpvdXRwdXQudmFsdWUgKz0gYG15X2Z1bmMoKSA9PiAke2V4cG9ydHMubXlfZnVuYygpfVxuYAo8L3NjcmlwdD4=)

Probably `wasmtime` use some hard limits for max memory by default?

**Comment by phenax** [comment]
The example is running wasm in the browser and I'm using it with wasi. I tried setting the memory to 60,000 pages and it still gives me the error. I've tried the exact same process with a wasm module built using rust (wasm32-wasi target) and that seems to be working fine with just the default 17 pages of memory.

And it is pretty weird for that particular expression to error out because if the function only contains a `console.log("whatever")`, everything seems to be working fine.
